source/common/upstream/thread_aware_lb_impl.cc

```diff
@@ -31,9 +38,16 @@ void normalizeLocalityWeights(const HostsPerLocality& hosts_per_locality,
                               double& min_normalized_weight, double& max_normalized_weight) {
   ASSERT(locality_weights.size() == hosts_per_locality.get().size());
 
-  uint32_t sum = 0;
+  // sum should be at most uint32_t max value, so we can validate it by accumulating into unit64_t
+  // and making sure there was no overflow
+  uint64_t sum = 0;
   for (const auto weight : locality_weights) {
     sum += weight;
+    if (sum > std::numeric_limits<uint32_t>::max()) {
+      throw EnvoyException(
+          fmt::format("Load balancing locality weights has accumulated over its maximum value {}",
```

Comment:
I think this error message could be a bit clearer: it might not be obvious to an operator that these weights are being accumulated. Maybe the message should be around the fact that the sum of locality weights within a priority causing overflow? Similar with host weights, though I'm not sure what the invariant is there (per locality? per priority?)